### DISCUSSION TITLE
How to use update-snapshot

### DISCUSSION THREAD

**Discussion by rethab** [origin-energy/java-snapshot-testing#174] [General]
Hi, according to the readme, I'm supposed to use `update-snapshot=none` in CI. At the same time, I want to use `update-snapshot=all` (I think) locally. I believe, that's the behavior I have now, but not specifying this property at all.

However, I've started receiving warnings and I'm not sure what to do. Based on the readme, I probably need two config files? Or is there another way to maintain the same behavior as before?

**Comment by jackmatt2** [comment]
Just add an environment variable on your build system 

```sh
export CI=true
```

**Reply by rethab** [reply]
I'm pretty sure this variable should be set correctly in our CI system, that's not my issue.

My issue, presupposing based on your comment, that the readme says:

> [none]=update no snapshots
> [MyTest1,MyTest2]=update snapshots in these classes only
>
> *Note: must be set to [none] on CI

That is, it's saying I need to set this to a specific value in CI. But are you saying that's actually not necessary assuming `CI=true` is correctly set?

**Reply by jackmatt2** [reply]
Yes, CI variable should override - I can update the docs to make it clearer

https://github.com/origin-energy/java-snapshot-testing/blob/master/java-snapshot-testing-core/src/main/java/au/com/origin/snapshots/SnapshotContext.java#L104C16-L104C20

**Reply by rethab** [reply]
ah perfect! Thanks :)

**Comment by jonathanberthias** [comment]
Hi @jackmatt2,

I tried to set `update-snapshot=all` but when running in CI, I got this error:
```
au.com.origin.snapshots.exceptions.SnapshotExtensionException: isCI=true & update-snapshot=Optional[]. Updating snapshots on CI is not allowed
```
What am I missing to get things to work as expected?

**Reply by jackmatt2** [reply]
Yeah, this is by design.  The snapshots should never update on CI or else they will always pass.

**Reply by jackmatt2** [reply]
If you really have a use case for updating snapshots on CI.  Try running the following
```
export CI=false
```
